drivers: ethernet: esp32: unused driver static function when ref_clk_output_gpios is not used

### Describe the bug


Building Zephyr for ESP32 Ethernet boards variant may fail with the following warning treated as an error:

`eth_esp32.c.obj.d -o zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_esp32.c.obj -c /__w/zephyr/zephyr/drivers/ethernet/eth_esp32.c /__w/zephyr/zephyr/drivers/ethernet/eth_esp32.c:245:13: error: 'eth_esp32_iomux_rmii_clk_input' defined but not used [-Werror=unused-function] 245 | static void eth_esp32_iomux_rmii_clk_input(void) |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ `

### Regression

- [x] This is a regression.

### Steps to reproduce

1. Use any ESP32 board with ethernet without external ref clk gpio for eth clock.
2. Build the project with warnings treated as errors.

### Relevant log output

```shell
eth_esp32.c.obj.d -o zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_esp32.c.obj -c /__w/zephyr/zephyr/drivers/ethernet/eth_esp32.c /__w/zephyr/zephyr/drivers/ethernet/eth_esp32.c:245:13: error: 'eth_esp32_iomux_rmii_clk_input' defined but not used [-Werror=unused-function] 245 | static void eth_esp32_iomux_rmii_clk_input(void) |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

OS: Linux
Toolchain: Zephyr SDK 1.0.0
Commit SHA or Version used: affected current main branch

### Additional Context

_No response_

--- a/drivers/ethernet/eth_esp32.c
+++ b/drivers/ethernet/eth_esp32.c
@@ -242,6 +242,7 @@ static uint32_t eth_esp32_receive_frame(struct eth_esp32_dev_data *dev_data, uin
 	return copy_len;
 }
 
+#if !DT_INST_NODE_HAS_PROP(0, ref_clk_output_gpios)
 static void eth_esp32_iomux_rmii_clk_input(void)
 {
 	const emac_iomux_info_t *pin = emac_rmii_iomux_pins.clki;
@@ -252,6 +253,7 @@ static void eth_esp32_iomux_rmii_clk_input(void)
 		PIN_FUNC_SELECT(GPIO_PIN_MUX_REG[pin->gpio_num], pin->func);
 	}
 }
+#endif
 
 static void eth_esp32_iomux_init_mii(void)
 {

